docs/modules/ROOT/examples/defaults_test.go

```diff
@@ -0,0 +1,34 @@
+package main
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+var (
+	testPath = "../../compiled/espejo/espejo"
+)
+
+func Test_Deployment_DefaultParameters(t *testing.T) {
+
+	subject := DecodeDeployment(t, testPath+"/10_deployment.yaml")
+	container := subject.Spec.Template.Spec.Containers[0]
```

Comment:
Hm, I disagree on this one. Having to assert each field just because it could be nil makes the unit tests super verbose, unreadable and painful to maintain. Here I sacrifice error readability for more readable unit test, since we are doing whitebox testing anyway.
Unfortunately asserting the "path" of this tree is hard in Go...

May I suggest a different approach with a Rego policy? e.g.
```
deny_deployment[msg] {
  input.kind == "Deployment"
  not input.spec.template.spec.containers[0]
  msg = "Deployment does not have a container"
}
```

Reply:
The fields can't be `nil` since they're not a pointer type. So worst case they're just the default value which would be catched by this test. Only thing is the `Containers` array which could be empty.

Reply:
Ah, that is good to know. So then, having an assertion for the array not to be empty should be simple enough.

Reply:
I can do that for this case.

The point still stands though. What do we do when there is a pointer? e.g. `resources` and the volumes/mounts. This example in the tutorial doesn't need that per se. Resources aren't validated in the test yet even though Espejo has defined some default resources, right now we're just showcasing how assertions could be done based on a simple MWE. 

OTOH, we could also simply leave that question to the component author(s).

Reply:
> OTOH, we could also simply leave that question to the component author(s).

For now yes. Later we might come up with a collection of best practices and or a helper library.